### PR TITLE
feat: enable to create aws_iam_openid_connect_provider.github in module

### PR DESCRIPTION
**Pull request by tk3fftk**
I think This module should be able to create OIDC provider because it has already adopted OIDC provider implicitly. 
This PR provide to enable to create OIDC provider in module. Some user may already have OIDC provider so it's optional and default is false.

ref: https://zenn.dev/shunsuke_suzuki/articles/tfaction-setup#iam-openid-connect-provider-%E3%81%AE%E4%BD%9C%E6%88%90

### CODE REVIEW

**Comment by suzuki-shunsuke** [comment]
Oh, sorry. I forgot this pr.
I'll take a look.
I'm wondering if `thumbprint_list` can be an empty list.

File: oidc-provider.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+resource "aws_iam_openid_connect_provider" "github" {
+  count = var.create_oidc_provider ? 1 : 0
+
+  url = "https://token.actions.githubusercontent.com"
+  # This is not required for GitHub Action OIDC integration with AWS but thumbprint_list is required so it pinned to the latest thumbprint
+  # ref: https://github.blog/changelog/2023-07-13-github-actions-oidc-integration-with-aws-no-longer-requires-pinning-of-intermediate-tls-certificates/
+  thumbprint_list = [
+    "d89e3bd43d5d909b47a18977aa9d5ce36cee184c",
```
Comment (suzuki-shunsuke) [inline oidc-provider.tf:8]: If we need to update the thumbprint periodically, it's bothersome.
I'm wondering if we can set an empty list or a dummy value.

`tls_certificate` is useful, but to use this resource we need to add the dependency on tls provider.
This is undesirable.

```tf
data "tls_certificate" "github" {
  url = "https://token.actions.githubusercontent.com/.well-known/openid-configuration"
}

resource "aws_iam_openid_connect_provider" "github" {
  url             = "https://token.actions.githubusercontent.com"
  thumbprint_list = [data.tls_certificate.github.certificates[0].sha1_fingerprint]
  client_id_list  = ["sts.amazonaws.com"]
}
```

Comment (suzuki-shunsuke) [reply oidc-provider.tf:8]: Maybe lifecycle.ignore_changes is useful to ignore the resource change when we import existing resource.

https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#ignore_changes

```tf
resource "aws_iam_openid_connect_provider" "github" {
  url             = "https://token.actions.githubusercontent.com"
  thumbprint_list = ["d89e3bd43d5d909b47a18977aa9d5ce36cee184c"]
  client_id_list  = ["sts.amazonaws.com"]

  lifecycle {
    ignore_changes = [
      thumbprint_list,
    ]
  }
}
```

**Comment by suzuki-shunsuke** [comment]
> I'm wondering if thumbprint_list can be an empty list.

I've confirmed terraform apply fails if `thumberprint_list` is empty.

```
╷
│ Error: error creating IAM OIDC Provider: InvalidInput: Thumbprint list must contain at least one entry.
│ 	status code: 400, request id: ***
│ 
│   with aws_iam_openid_connect_provider.github,
│   on main.tf line 65, in resource "aws_iam_openid_connect_provider" "github":
│   65: resource "aws_iam_openid_connect_provider" "github" {
│ 
╵
```

**Comment by suzuki-shunsuke** [comment]
v0.2.2 is out 🎉
https://github.com/suzuki-shunsuke/terraform-aws-tfaction/releases/tag/v0.2.2

**Comment by tk3fftk** [comment]
Sorry, thanks!!

**Pull request merged**